tests/track_errors/test_ctc_errors.py

```diff
@@ -194,15 +272,257 @@ def test_ns_vertex_fn_edge():
+            assert attrs.get(EdgeFlag.INTERTRACK_EDGE) is False
+
+        for attrs in matched.pred_graph.edges.values():
+            assert attrs.get(EdgeFlag.CTC_FALSE_POS) is False
+            assert attrs.get(EdgeFlag.INTERTRACK_EDGE) is False
+            assert attrs.get(EdgeFlag.WRONG_SEMANTIC) is False
+
+    def test_fn_node_end(self):
+        matched = self.prep_matched(ex_graphs.fn_node_matched(0))
+
+        # All pred edges correct
+        for attrs in matched.pred_graph.edges.values():
+            assert attrs.get(EdgeFlag.CTC_FALSE_POS) is False
+            assert attrs.get(EdgeFlag.INTERTRACK_EDGE) is False
+            assert attrs.get(EdgeFlag.WRONG_SEMANTIC) is False
+
+        # First gt edge is false neg
+        attrs = matched.gt_graph.edges[(1, 2)]
+        assert attrs.get(EdgeFlag.CTC_FALSE_NEG) is True
+        assert attrs.get(EdgeFlag.INTERTRACK_EDGE) is False
+
+    def test_fn_node_middle(self):
+        matched = self.prep_matched(ex_graphs.fn_node_matched(1))
+
+        # No pred edges to test
+
+        # All gt edges false pos
+        for attrs in matched.gt_graph.edges.values():
+            assert attrs.get(EdgeFlag.CTC_FALSE_NEG) is True
+            assert attrs.get(EdgeFlag.INTERTRACK_EDGE) is False
+
+    def test_fn_edge(self):
+        matched = self.prep_matched(ex_graphs.fn_edge_matched(0))
+
+        # Only pred edge is correct
+        attrs = matched.pred_graph.edges[(5, 6)]
+        assert attrs.get(EdgeFlag.CTC_FALSE_POS) is False
+        assert attrs.get(EdgeFlag.INTERTRACK_EDGE) is False
+
+        # First gt edge is false neg
+        attrs = matched.gt_graph.edges[(1, 2)]
+        assert attrs.get(EdgeFlag.CTC_FALSE_NEG) is True
+        assert attrs.get(EdgeFlag.INTERTRACK_EDGE) is False
+
+        # Second gt edge is correct
+        attrs = matched.gt_graph.edges[(2, 3)]
+        assert attrs.get(EdgeFlag.CTC_FALSE_NEG) is False
+        assert attrs.get(EdgeFlag.INTERTRACK_EDGE) is False
+
+    @pytest.mark.parametrize("t", [0, 1, 2])
+    def test_fp_node(self, t):
+        matched = self.prep_matched(ex_graphs.fp_node_matched(t))
+
+        # All pred edges correct
+        for attrs in matched.pred_graph.edges.values():
+            assert attrs.get(EdgeFlag.CTC_FALSE_POS) is False
+            assert attrs.get(EdgeFlag.INTERTRACK_EDGE) is False
+            assert attrs.get(EdgeFlag.WRONG_SEMANTIC) is False
+
+        # All gt edges correct
+        for attrs in matched.gt_graph.edges.values():
+            assert attrs.get(EdgeFlag.CTC_FALSE_NEG) is False
+            assert attrs.get(EdgeFlag.INTERTRACK_EDGE) is False
+
+    @pytest.mark.parametrize("t", [0, 1])
+    def test_fp_edge(self, t):
+        matched = self.prep_matched(ex_graphs.fp_edge_matched(t))
+
+        # All but one pred edges correct
+        for edge, attrs in matched.pred_graph.edges.items():
+            # false positive edge
+            if edge == (7, 8):
```

Comment:
The if and else branches here are identical (because the FP edge is entirely made of FP nodes), so maybe we should remove this test and keep the one below that does have FP edges involved?

Reply:
It's a little redundanct, but I'd actually like to keep this in here just for my own understanding/clarity. This case of fp edge is unintuitive to me so it's helpful to have an explicit assertion of the correct behavior. Caroline and I are also trying to have every error flagging method test against all available test cases, so this maintains that coverage.

I'll add to the comment so that it's clear why we are bothering.